src/Microsoft.AspNet.Mvc.Core/Rendering/HtmlHelperOptions.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNet.Mvc.Rendering
+{
+    /// <summary>
+    /// Provides programmatic configuration for the HTML helpers and <see cref="ViewContext"/>.
+    /// </summary>
+    public class HtmlHelperOptions
+    {
+        public HtmlHelperOptions()
+        {
+            IdAttributeDotReplacement = "_";
+            ClientValidationEnabled = true;
+            ValidationMessageElement = "span";
+            ValidationSummaryMessageElement = "span";
+        }
+
+        /// <summary>
+        /// Gets or sets the <see cref="Html5DateRenderingMode.Html5DateRenderingMode"/> value.
+        /// </summary>
+        /// <remarks>
+        /// Set this property to <see cref="Html5DateRenderingMode.Rfc3339"/> to have templated helpers such as
+        /// <see cref="IHtmlHelper.Editor"/> and <see cref="IHtmlHelper{TModel}.EditorFor"/> render date and time 
+        /// values as RFC 3339 compliant strings. By default these helpers render dates and times using the current 
+        /// culture.
+        /// </remarks>
+        public Html5DateRenderingMode Html5DateRenderingMode { get; set; }
+
+        /// <summary>
+        /// Gets or sets the <see cref="string"/> that replaces periods in the ID attribute of an element.
+        /// </summary>
+        public string IdAttributeDotReplacement { get; set; }
+
+        /// <summary>
+        /// Gets or sets a value that indicates whether client-side validation is enabled.
+        /// </summary>
+        public bool ClientValidationEnabled { get; set; }
```

Comment:
@Eilon agreed.  The other ones are not the same as `IdAttributeDotReplacement` in either MVC 5 or 6.  `ViewContext` already did something that would feel much like the old `ScopedStorage`-based feature from MVC 5.  So we're providing the old (super secret) feature and the public surface isn't all that different.